### DISCUSSION TITLE
Live microphone with auto language?

### DISCUSSION THREAD

**Discussion by adriangalilea** [deepgram/community#271] [General help] [answered]
Hello, I've been using OpenAI API for some months and I'm very happy with the results, particularly hot-swapping languages which is a common occurence in any conversation as specially talking about tech even in spanish you end up saying tons of english words.

- I need multi-language support.
- I need streaming support.

OpenAI doesn't give me any option for streaming so I've been trying deepgram, my problem is:
If I don't set language it defaults to english and none is picked up.

But if I set:
`&language=es&tier=nova`

It can barely pick up my name and saying Python is outright impossible:
```
Hola.
Soy Arian. Soy Arian.
Soy Adrián.
Garilea,
Galilea.
Estoy escribiendo una aplicación en Piphone,
Estoy escribiendo una app
en Paizon,
Pyzon
```

Is there anything I can do to improve this results?

Thank you.

**Comment by jkroll-deepgram** [comment, marked as answer]
Hi @adriangalilea, you're correct to specify a language for streaming. [Language detection](https://developers.deepgram.com/docs/language-detection) is only supported with pre-recorded audio, not streaming.

It appears that the words that are causing difficulty are proper nouns (your name, Python, iPhone). These tend to be less accurate than regular words, as they occur less frequently. If you notice a pattern with a certain frequent proper noun, you can use the [Find & Replace](https://developers.deepgram.com/docs/find-and-replace) feature. For instance, if you're continually seeing "Adrian" transcribed as "Arian", you can specify the parameter `replace=arian:Adrian`, which will replace all instances of "Arian" with "Adrian" in the transcript (note that the find-term is lowercase and the replace-term is the case in which you want it to appear).

**Reply by adriangalilea** [reply]
I see, not practical for my use case, but I'm glad I understand the limitations now.

Thank you.
